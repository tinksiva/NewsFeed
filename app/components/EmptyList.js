import { StyleSheet, Text, View } from "react-native";
import { memo } from "react";

function EmptyList(props) {
  return (
    <View style={styles.emptyComponentStyles}>
      <Text style={styles.emptyText}>
        {"Please Wait while we fetch the latest news for you"}
      </Text>
    </View>
  );
}

const styles = StyleSheet.create({
  emptyComponentStyles: {
    justifyContent: "center",
    alignItems: "center",
    alignSelf: "center",
    width: "100%",
    height: "100%",
    flex: 1,
  },
  emptyText: { textAlign: "center", alignSelf: "center", marginTop: "50%" },
});

export default memo(EmptyList);
