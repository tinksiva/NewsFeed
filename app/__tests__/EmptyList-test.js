jest.useFakeTimers()
import 'react-native';
import React from 'react';
import EmptyList from "../components/EmptyList";

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <EmptyList />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});