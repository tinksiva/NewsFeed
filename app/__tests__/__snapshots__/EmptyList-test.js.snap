// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "alignSelf": "center",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "width": "60%",
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "center",
        "textAlign": "center",
      }
    }
  >
    Please Wait while we fetch the latest news for you
  </Text>
</View>
`;
